spec/app/domain/authentication/resource_restrictions/get_service_specific_restriction_from_annotation_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe(Authentication::ResourceRestrictions::GetServiceSpecificRestrictionsFromAnnotation) do
+  let(:authenticator_name) { 'authn-dummy' }
+  let(:service_name) { 'dummy_service' }
+  let(:restriction_name) { 'user_email' }
+  let(:general_annotation_name) { "#{authenticator_name}/#{restriction_name}" }
+  let(:service_specific_annotation_name) { "#{authenticator_name}/#{service_name}/#{restriction_name}" }
+
+  #  ____  _   _  ____    ____  ____  ___  ____  ___
+  # (_  _)( )_( )( ___)  (_  _)( ___)/ __)(_  _)/ __)
+  #   )(   ) _ (  )__)     )(   )__) \__ \  )(  \__ \
+  #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
+
+  context "GetServiceSpecificRestrictionsFromAnnotation" do
```

Comment:
same as the comment above please add more tests

Reply:
Fixed